app/views/SettingsView/index.js

```diff
@@ -1,236 +1,120 @@
-				label: 'Français',
-				value: 'fr'
-			}, {
-				label: 'Deutsch',
-				value: 'de'
-			}, {
-				label: 'Português (PT)',
-				value: 'pt-PT'
-			}],
-			saving: false
-		};
+	toggleMarkdown = (value) => {
+		AsyncStorage.setItem(MARKDOWN_KEY, JSON.stringify(value));
+		const { toggleMarkdown } = this.props;
+		toggleMarkdown(value);
+		firebase.analytics().logEvent.logCustom('toggle_markdown', { value });
 	}
 
-	shouldComponentUpdate(nextProps, nextState) {
-		const { language, saving } = this.state;
-		const { userLanguage, useMarkdown } = this.props;
-		if (nextState.language !== language) {
-			return true;
-		}
-		if (nextState.saving !== saving) {
-			return true;
-		}
-		if (nextProps.useMarkdown !== useMarkdown) {
-			return true;
-		}
-		if (nextProps.userLanguage !== userLanguage) {
-			return true;
-		}
-		return false;
+	navigateToRoom = (room) => {
+		const { navigation } = this.props;
+		return () => navigation.navigate(room);
 	}
 
-	getLabel = (language) => {
-		const { languages } = this.state;
-		const l = languages.find(i => i.value === language);
-		if (l && l.label) {
-			return l.label;
+	sendEmail = () => {
+		const subject = encodeURI('React Native App Support');
+		const email = encodeURI('support@rocket.chat');
+		const description = encodeURI(`
+			version: ${ getReadableVersion }
+			device: ${ getDeviceModel }
+		`);
+		try {
+			Linking.openURL(`mailto:${ email }?subject=${ subject }&body=${ description }`);
+		} catch (e) {
+			showErrorAlert(I18n.t('error-email-send-failed', { action: 'support@rocket.chat' }));
 		}
-		return null;
-	}
-
-	formIsChanged = () => {
-		const { userLanguage } = this.props;
-		const { language } = this.state;
-		return !(userLanguage === language);
 	}
 
-	submit = async() => {
-		this.setState({ saving: true });
-
-		const { language } = this.state;
-		const { userLanguage, setUser } = this.props;
-
-		if (!this.formIsChanged()) {
-			return;
-		}
-
-		const params = {};
+	openLink = link => openLink(link)
 
-		// language
-		if (userLanguage !== language) {
-			params.language = language;
-		}
+	renderSectionSeparator = () => <View style={styles.sectionSeparatorBorder} />;
 
-		try {
-			await RocketChat.saveUserPreferences(params);
-			setUser({ language: params.language });
-
-			this.setState({ saving: false });
-			setTimeout(() => {
-				this.toast.show(I18n.t('Preferences_saved'));
-			}, 300);
-		} catch (e) {
-			this.setState({ saving: false });
-			setTimeout(() => {
-				showErrorAlert(I18n.t('There_was_an_error_while_action', { action: I18n.t('saving_preferences') }));
-				log('err_save_user_preferences', e);
-			}, 300);
-		}
-	}
+	renderSeparator = () => <View style={styles.separator} />;
 
-	toggleMarkdown = (value) => {
-		AsyncStorage.setItem(MARKDOWN_KEY, JSON.stringify(value));
-		const { toggleMarkdown } = this.props;
-		toggleMarkdown(value);
-		firebase.analytics().logEvent('toggle_markdown', { value });
-	}
+	renderCrashReportDisclaimer = () => (
```

Comment:
Remove it.

Reply:
I'll be commenting this as it will be needed in the future